### DISCUSSION TITLE
How to look at the underlying technology(如何看待底层技术)

### DISCUSSION THREAD

**Discussion by JanYork** [community/community#29649] [Other Feature Feedback, Questions, & Ideas]
我是一个Java程序员，不是很喜欢底层技术，喜欢应用层，但是许许多多技术总会要深入到底层原理，但是我发现，在正常的开发中，我更多是使用框架，而非开发框架或者去深入框架原理，各位大佬作为应用层程序员(JavaWeb，GoWeb，Vue)，如何看待对底层原理的深入学习，也没有必要。可能日常开发SpringBoot、Kafka、mq这类框架只需要会使用就可以，但是看到许多人对底层都很深入。

English：
I am a Java programmer, I don't like the underlying technology very much, I like the application layer, but many technologies will always go deep into the underlying principles, but I found that in normal development, I use frameworks more than development Framework or go deep into the framework principle, as application layer programmers (JavaWeb, GoWeb, Vue), how do you think about the in-depth study of the underlying principles, it is not necessary. It may be possible to develop frameworks such as SpringBoot, Kafka, and mq on a daily basis. However, many people have a deep understanding of the underlying layers.

**Comment by PUSHPAK-JAISWAL** [comment]
Well it depends on the Goal if your current goal is to become a application developer that gets work done you are pretty much there.

Now if you are some what of a person trying to get a job then there is no telling that the person Infront of you might ask you deep or logical questions so for that time you need to be prepared at least with the basic understanding how internal things work to land a job.

Now once you have joined a MNC with a lot's of reputation like some company from MAANG you could need to learn about internal things in deep.

Why, let me tell you, if you are a developer you know react right so react was a internal tool developed by meta (at time Facebook), so you understand the companies are always trying to keep things internal for some other reason so to make and use internal tools you might need in depth knowledge. Because companies make internal tools to to take edge in the software the have encountered problems with to keep there systems alive.

so, it depends on goal
fresher :- just framework is enough
intermediate :- some knowledge
Elite :- prefect internal knowledge is required
